packages/smooth_app/lib/pages/offline_data_page.dart

```diff
@@ -0,0 +1,241 @@
+                      ),
+                      duration: SnackBarDuration.brief,
+                    ),
+                  );
+                }
+              },
+            ),
+            _OfflinePageListTile(
+              title: 'Clear Offline Product Data',
+              subtitle:
+                  'Clear all local product data from your app to free up space',
+              trailing: const Icon(Icons.delete),
+              onTap: () async {
+                final int totalProductsDeleted = await daoProduct.deleteAll();
+                if (mounted) {
+                  ScaffoldMessenger.of(context).showSnackBar(
+                    SnackBar(
+                      content: Text('$totalProductsDeleted products deleted'),
+                      duration: SnackBarDuration.brief,
+                    ),
+                  );
+                }
+                setState(() {});
+              },
+            ),
+            _OfflinePageListTile(
+              title: 'Know More',
+              subtitle: 'Click to know more about offline data',
+              trailing: const Icon(Icons.info),
+              // ignore: avoid_returning_null_for_void
+              onTap: () => null,
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+// Widget to display the stats of the local databas, ie. the number of products
+// in the database and the size of the database
+class _StatsWidget extends StatelessWidget {
+  const _StatsWidget({
+    Key? key,
+    required this.daoProduct,
+  }) : super(key: key);
+  final DaoProduct daoProduct;
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(vertical: SMALL_SPACE),
+      child: ListTile(
+        title: const Text('Offline Product Data'),
+        subtitle: FutureBuilder<int>(
+          future: daoProduct.getTotalNoOfProducts(),
+          builder: (BuildContext context, AsyncSnapshot<int> snapshot) {
+            if (snapshot.hasData) {
+              return Text(
+                  '${snapshot.data} products available for immediate scaning');
+            } else {
+              return const Text('0 products available for immediate scaning');
+            }
+          },
+        ),
+        trailing: FutureBuilder<double>(
+          future: daoProduct.getEstimatedTotalSizeInMB(),
+          builder: (BuildContext context, AsyncSnapshot<double> snapshot) {
+            if (snapshot.hasData) {
+              return Text('${snapshot.data} MB');
+            } else {
+              return const SizedBox(
+                height: 20,
+                width: 20,
+                child: CircularProgressIndicator.adaptive(),
+              );
+            }
+          },
+        ),
+      ),
+    );
+  }
+}
+
+// Widget to display a list tile with a title, subtitle
+// and a trailing widget and an onTap callback for OfflineDataPage
+class _OfflinePageListTile extends StatelessWidget {
```

Comment:
We at some point should unify the settings but thats okay for now

Reply:
Yup, and i might need to refactor it differently but i guess it's good for nwo